.gitignore

```diff
@@ -9,3 +9,6 @@ compile_commands.json
 .vscode/
 *.log
 *.csv
+
+# Generated flatbuffer header for simulation device
+device/simulation/tt_simulation_device_generated.h
```

Comment:
nit comment here. Maybe it would be nicer to have `device/simulation/generated/` directory, so we can gitignore whole dir. This would be bulletproof for future generated files. Not requesting this to be changed, you can consider it if you want. If not, I am totally fine with that as well

Reply:
Woops, I changed this to generate into the build folder, so this .gitignore change is not needed anymore. Will change.

Reply:
fixed